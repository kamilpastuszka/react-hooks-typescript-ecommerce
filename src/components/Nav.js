import React from "react";

export default function Nav() {
  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-light bg-light align-content-between">
        <div className="navbar-brand">ecommerce app</div>
        <div className="collapse navbar-collapse" id="navbarColor02" />
        <div className="pull-right">basket </div>
      </nav>
    </>
  );
}
