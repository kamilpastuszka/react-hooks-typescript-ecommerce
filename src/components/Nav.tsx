import React, { useContext } from "react";
import { FaShoppingCart, FaStore } from "react-icons/fa";
import { Link } from "react-router-dom";
import { Context } from "../context/context";

export default function Nav(props: any) {
  const { state } = useContext(Context);

  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-light bg-light align-content-between">
        <div className="navbar-brand">ecommerce app</div>
        <div className="collapse navbar-collapse" id="navbarColor02" />
        <div className="pull-right align-items-center">
          <Link to="/">
            <span className="mr-3">
              {" "}
              store <FaStore />{" "}
            </span>
          </Link>

          <Link to="/Basket">
            <span>
              {" "}
              basket <FaShoppingCart />
              <span className="itemCount">{state.length}</span>
            </span>
          </Link>
        </div>
      </nav>
    </>
  );
}
