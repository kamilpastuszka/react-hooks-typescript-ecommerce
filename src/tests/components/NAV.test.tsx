import React from "react";
import { shallow } from "enzyme";
import Nav from "../../components/Nav";
import toJson from "enzyme-to-json";

test("should render Basket correctly", () => {
  const wrapper = shallow(<Nav />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
