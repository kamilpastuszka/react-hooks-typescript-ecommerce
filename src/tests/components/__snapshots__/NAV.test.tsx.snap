// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Basket correctly 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light align-content-between"
  >
    <div
      className="navbar-brand"
    >
      ecommerce app
    </div>
    <div
      className="collapse navbar-collapse"
      id="navbarColor02"
    />
    <div
      className="pull-right align-items-center"
    >
      <Link
        to="/"
      >
        <span
          className="mr-3"
        >
           
          store 
          <FaStore />
           
        </span>
      </Link>
      <Link
        to="/Basket"
      >
        <span>
           
          basket 
          <FaShoppingCart />
          <span
            className="itemCount"
          >
            1
          </span>
        </span>
      </Link>
    </div>
  </nav>
</Fragment>
`;
